### PR TITLE
Update SoundWire interfaces

### PR DESCRIPTION
**Pull request by plbossart**
These two files are extracted from PR #1656 and should be merged first, before adding the functionality.

### CODE REVIEW

File: include/sound/soc-acpi.h [hunk 1]
```diff
@@ -61,13 +61,32 @@ static inline struct snd_soc_acpi_mach *snd_soc_acpi_codec_list(void *arg)
...
+ * @links: array of link _ADR descriptors, null terminated
  */
 struct snd_soc_acpi_mach_params {
 	u32 acpi_ipc_irq_index;
 	const char *platform;
 	u32 codec_mask;
 	u32 dmic_num;
 	bool common_hdmi_codec_drv;
+	u32 link_mask;
+	const struct snd_soc_acpi_link_adr *links;
```
Comment (ranj063) [inline include/sound/soc-acpi.h:74]: @plbossart what is the rationale for having this array of links in both the acpi_mach and mach_params?

Comment (plbossart) [reply include/sound/soc-acpi.h:74]: these values are needed for the acpi_mach to select a machine driver/topology.

They are also needed within the machine driver to select which dailinks need to be enabled if the same machine driver is used for multiple machines.

**Comment by plbossart** [comment]
merging since it's needed for SoundWire development. We can always update those structures before submitting upstream.

File: include/sound/soc-acpi.h (hunk 1 above)
Comment (ranj063) [reply include/sound/soc-acpi.h:74]: @plbossart I see so the problem is that only mach_params are passed to the machine driver? Can we use the container_of to get the acpi_mach and then access the links from there? How do you use it today? Do you point the links in the mach_params struct to the links in the acpi_mach?

Comment (plbossart) [reply include/sound/soc-acpi.h:74]: Yes, I thought about this. 

the acpi_mach structure is mostly made of constant fields (except for pdata).
The mach_params are mostly dynamic today (codec mask, dmic num, etc)

Here it's mixed. the link mask could also be dynamic and updated with a kernel parameter, but the link descriptors are probably static. we may be able to use container_of indeed if we always reuse know values. I prefer to punt and see later if we can simplify. the way by which we pass the parameters doesn't really matter, it's the access to the information that helps.

**Pull request merged**